File: frontend/src/types/portfolioConfig.ts
/**
 * Portfolio configuration entity.
 */
export interface PortfolioConfig {
  id: number;
  name: string;
  portfolio_strategy: string;
  position_size: number;
  min_buy_score: number;
  trailing_stop_pct: number;
  max_per_sector: number | null;
  max_open_positions: number | null;
}

/**
 * Response from GET /api/v1/portfolio-configs.
 */
export interface PortfolioConfigListResponse {
  items: PortfolioConfig[];
  total: number;
}

/**
 * Request body for creating a new portfolio configuration.
 */
export interface CreatePortfolioConfigRequest {
  name: string;
  portfolio_strategy: string;
  position_size?: number;
  min_buy_score?: number;
  trailing_stop_pct?: number;
  max_per_sector?: number | null;
  max_open_positions?: number | null;
}

/**
 * Request body for updating a portfolio configuration.
 */
export interface UpdatePortfolioConfigRequest {
  name?: string;
  portfolio_strategy?: string;
  position_size?: number;
  min_buy_score?: number;
  trailing_stop_pct?: number;
  max_per_sector?: number | null;
  max_open_positions?: number | null;
}

File: backend/app/models/portfolio_config.py
"""Portfolio configuration model for named portfolio selection presets."""
from sqlalchemy import Float, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class PortfolioConfig(Base):
    """Named portfolio configuration preset.

    Stores reusable portfolio selection settings that can be selected
    when creating Arena simulations.
    """

    __tablename__ = "portfolio_configs"

    name: Mapped[str] = mapped_column(
        String(100),
        nullable=False,
        unique=True,
        doc="User-provided name (e.g., 'Conservative Sector Caps')",
    )
    portfolio_strategy: Mapped[str] = mapped_column(
        String(64),
        nullable=False,
        default="none",
        doc="Portfolio selector strategy key",
    )
    position_size: Mapped[int] = mapped_column(
        Integer,
        nullable=False,
        default=1000,
        doc="Per-position dollar allocation used for simulations",
    )
    min_buy_score: Mapped[int] = mapped_column(
        Integer,
        nullable=False,
        default=60,
        doc="Minimum score required for BUY decisions",
    )
    trailing_stop_pct: Mapped[float] = mapped_column(
        Float,
        nullable=False,
        default=5.0,
        doc="Trailing stop percentage used for simulations",
    )
    max_per_sector: Mapped[int | None] = mapped_column(
        Integer,
        nullable=True,
        doc="Max concurrent positions per sector (null = unlimited)",
    )
    max_open_positions: Mapped[int | None] = mapped_column(
        Integer,
        nullable=True,
        doc="Max total open positions (null = unlimited)",
    )
